src/v/model/panda_link.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "serde/envelope.h"
+#include "serde/rw/variant.h"
+#include "utils/named_type.h"
+#include "utils/unresolved_address.h"
+#include "utils/uuid.h"
+
+#include <seastar/util/bool_class.hh>
+
+namespace model {
+/// ID of the panda link - used internally based off of controller offset
+using panda_link_id = named_type<int64_t, struct panda_link_id_tag>;
+/// UUID of the panda link - used externally
+using panda_link_uuid = named_type<uuid_t, struct panda_link_uuid_tag>;
+/// Name of the panda link
+using panda_link_name = named_type<ss::sstring, struct panda_link_name_tag>;
+
+/**
+ * @brief SCRAM credentials to use for authentication
+ */
+struct scram_credentials
+  : serde::
+      envelope<scram_credentials, serde::version<0>, serde::compat_version<0>> {
+    /// SCRAM username
+    ss::sstring username;
+    /// SCRAM password
+    ss::sstring password;
+    /// SASL-SCRAM mechanism to use
+    ss::sstring mechanism;
+
+    friend bool operator==(const scram_credentials&, const scram_credentials&)
+      = default;
+    auto serde_fields() { return std::tie(username, password, mechanism); }
+};
+
+/**
+ * @brief Represents the settings for connection to a remote cluster
+ */
+struct panda_link_connection
```

Comment:
here the name suggest that the object represents connection, while it is only representing configuration, consider changing the name to `connection_configuration`